x-pack/plugins/canvas/public/services/kibana/data_views.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaPluginServiceFactory } from '@kbn/presentation-util-plugin/public';
+import { ErrorStrings } from '../../../i18n';
+import { CanvasStartDeps } from '../../plugin';
+import { CanvasDataViewsService } from '../data_views';
+import { CanvasNotifyService } from '../notify';
+
+const { esService: strings } = ErrorStrings;
+
+export type DataViewsServiceFactory = KibanaPluginServiceFactory<
+  CanvasDataViewsService,
+  CanvasStartDeps,
+  {
+    notify: CanvasNotifyService;
+  }
+>;
+
+export const dataViewsServiceFactory: DataViewsServiceFactory = ({ startPlugins }, { notify }) => ({
+  getDataViews: async () => {
+    try {
+      const dataViews = await startPlugins.data.dataViews.getIdsWithTitle();
+      return dataViews.map((view) => view.title);
```

Comment:
How is the list of data views being used? Title is no longer unique, and generally speaking data views should be referenced by `name` in UIs.

Reply:
<img width="345" alt="Screen Shot 2022-08-29 at 10 02 17 AM" src="https://user-images.githubusercontent.com/1697105/187244382-1eb47403-f04c-4408-8993-5dd709395690.png">

We use it to populate this dropdown menu, but I can certainly switch to using the IDs as the value and the `name` as the label. Let me update the PR.

Reply:
For BWC, I won't be able to use the data view `id` instead of the `title` because this value is used directly in the SQL statement generated in this `esdocs` datasource function and updates the SQL statement to something like `SELECT * FROM kibana_sample_data_flights`. Setting the value to `id` instead results in an error because ES doesn't recognize this ID and ES SQL expects an index name here. I can use `name` for the label for each of the options, but I'll still need to use `title` for the value selected and inserted into the SQL statement. 

I'm not sure that uniqueness is a requirement here because if you have two data views with the same title, ES will just return results from both data views right?

Reply:
Ouch, thats an awkward mixing of metaphors. This should definitely be resolved but I'm not sure how. Data views should be referenced by name even if their title is used for a query. 

Any idea how we might address this? Who owns SQL?

Reply:
Perhaps a short term fix would be adding more info to each item in the list - display the data view name but beneath it show the index pattern.

Reply:
> Any idea how we might address this? Who owns SQL?

@elastic/es-ql owns ES SQL

> Perhaps a short term fix would be adding more info to each item in the list - display the data view name but beneath it show the index pattern.

You mean display both the `title` and `name` for each data view in the dropdown list? Is there a problem with seeing the `name` only in the option list and seeing the `title` in the expression/SQL statement?

Reply:
>Is there a problem with seeing the name only in the option list and seeing the title in the expression/SQL statement?

Thats your call but I worry that it might confuse some.